### PR TITLE
[Model] Apply appcontext

### PR DESCRIPTION
**Pull request by zhoonit**
- ~**#728** [AppContext] Add AppContext~
- [Model] Apply appcontext
```
Apply appcontext to NeuralNetwork. From this patch `chdir()` hack is not
needed :)

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>
```

### CODE REVIEW

File: jni/Android.mk [hunk 1]
```diff
@@ -82,7 +82,8 @@ NNTRAINER_SRCS := $(NNTRAINER_ROOT)/nntrainer/models/neuralnet.cpp \
                   $(NNTRAINER_ROOT)/nntrainer/optimizers/sgd.cpp \
                   $(NNTRAINER_ROOT)/nntrainer/optimizers/optimizer_factory.cpp \
                   $(NNTRAINER_ROOT)/nntrainer/utils/util_func.cpp \
-                  $(NNTRAINER_ROOT)/nntrainer/utils/parse_util.cpp
+                  $(NNTRAINER_ROOT)/nntrainer/utils/parse_util.cpp \
+									$(NNTRAINER_ROOT)/nntrainer/app_context.cpp
```
Comment (jijoongmoon) [inline jni/Android.mk:86]: Check this tab

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: jni/Android.mk (hunk 1 above)
Comment (zhoonit) [reply jni/Android.mk:86]: Done

File: Applications/TransferLearning/Draw_Classification/jni/Android.mk [hunk 2]
```diff
@@ -62,7 +62,8 @@ LOCAL_LDLIBS := -llog
 
 LOCAL_SRC_FILES := main.cpp
 
-LOCAL_SHARED_LIBRARIES := capi-nntrainer app_utils
+# @todo add api for context
+LOCAL_SHARED_LIBRARIES := nntrainer capi-nntrainer app_utils
```
Comment (kparichay) [inline Applications/TransferLearning/Draw_Classification/jni/Android.mk:66]: capi-nntrainer depends on nntrainer. I dont think nntrainer needs to be specified here.
check line 38.

Comment (zhoonit) [reply Applications/TransferLearning/Draw_Classification/jni/Android.mk:66]: Appcontext is not part of the `capi-nntrainer`, so without this line, the app throws DSO missing error

File: nntrainer/models/model_loader.cpp [hunk 3]
```diff
@@ -384,11 +386,14 @@ int ModelLoader::loadFromIni(std::string ini_file, NeuralNetwork &model,
      * @note The order of backbones in the ini file defines the order on the
      * backbones in the model graph
      */
-    const char *backbone =
+    const char *backbone_path =
       iniparser_getstring(ini, (sec_name + ":Backbone").c_str(), unknown);
-    if (backbone == unknown) {
+
+    const std::string &backbone =
```
Comment (kparichay) [inline nntrainer/models/model_loader.cpp:392]: can be moved to inside else if line 397,

**Review by kparichay — APPROVED** [review]
LGTM

File: Applications/TransferLearning/Draw_Classification/jni/Android.mk (hunk 2 above)
Comment (kparichay) [reply Applications/TransferLearning/Draw_Classification/jni/Android.mk:66]: Thats quite strange.

**Pull request merged**